src/Cmfd.cpp

```diff
@@ -606,311 +539,140 @@ FP_PRECISION Cmfd::computeKeff(int moc_iteration) {
-  matrix_zero(_A, _num_cmfd_groups*(_num_cmfd_groups+4), _num_x*_num_y);
+  _A->clear();
+  _M->clear();
   
   /* Loop over cells */
-  #pragma omp parallel for private(value, volume, cell, row, material)
+  #pragma omp parallel for private(value, volume, cell, material)
   for (int y = 0; y < _num_y; y++) {
     for (int x = 0; x < _num_x; x++) {
 
       cell = y*_num_x + x;
       material = _materials[cell];
-      volume = _volumes[cell];
+      volume =_volumes->getValue(cell, 0);
 
       /* Loop over groups */
       for (int e = 0; e < _num_cmfd_groups; e++) {
           
-        row = cell*_num_cmfd_groups + e;
-    
         /* Absorption term */
         value = material->getSigmaA()[e] * volume;
-        _A[cell][e*(_num_cmfd_groups+4)+e+2] += value;
+        _A->incrementValue(cell, e, cell, e, value);
         
         /* Out (1st) and in (2nd) scattering */
         for (int g = 0; g < _num_cmfd_groups; g++) {
           if (e != g) {
             value = material->getSigmaS()[g*_num_cmfd_groups + e] * volume;
-            _A[cell][e*(_num_cmfd_groups+4)+e+2] += value;
+            _A->incrementValue(cell, e, cell, e, value);
             value = - material->getSigmaS()[e*_num_cmfd_groups + g] * volume;
-            _A[cell][e*(_num_cmfd_groups+4)+g+2] += value;
+            _A->incrementValue(cell, g, cell, e, value);
           }
         }
 
-        /* RIGHT SURFACE */
+        /* SURFACE_X_MIN */
 
         /* Set transport term on diagonal */
-        value = (material->getDifHat()[2*_num_cmfd_groups + e]
-                - material->getDifTilde()[2*_num_cmfd_groups + e])
+        value = (material->getDifHat()[e] + material->getDifTilde()[e])
           * _cell_height;
-        
-        _A[cell][e*(_num_cmfd_groups+4)+e+2] += value;
+        _A->incrementValue(cell, e, cell, e, value);        
 
         /* Set transport term on off diagonal */
-        if (x != _num_x - 1) {
-          value = - (material->getDifHat()[2*_num_cmfd_groups + e]
-                  + material->getDifTilde()[2*_num_cmfd_groups + e])
-                  * _cell_height;
-            
-          _A[cell][e*(_num_cmfd_groups+4)+_num_cmfd_groups+2] += value;
+        if (x != 0) {
+          value = - (material->getDifHat()[SURFACE_X_MIN*_num_cmfd_groups + e]
+                     - material->getDifTilde()
+                     [SURFACE_X_MIN*_num_cmfd_groups + e]) * _cell_height;
+          _A->incrementValue(cell-1, e, cell, e, value);          
         }
 
-        /* LEFT SURFACE */
+        /* SURFACE_X_MAX */
 
         /* Set transport term on diagonal */
-        value = (material->getDifHat()[e]
-                + material->getDifTilde()[e])
-            * _cell_height;
-        
-
-        _A[cell][e*(_num_cmfd_groups+4)+e+2] += value;
+        value = (material->getDifHat()[SURFACE_X_MAX*_num_cmfd_groups + e]
+                - material->getDifTilde()[SURFACE_X_MAX*_num_cmfd_groups + e])
+          * _cell_height;
```

Comment:
I was just using the default emacs indent for continuation lines, which seems to indent statements within parentheses to the start of the parentheses and indent statements not in parentheses by two spaces (one tab). As far as I can see, the Google C++ style guide is ambiguous on line continuation. On continuation of function calls, the Google C++ style guide suggests to indent statements within parentheses to the start of the parentheses or by four spaces (two tabs).

I'm fine with any format that looks readable. The current code base indents function calls to the first argument. The majority of line continuations seem to be from code that I pushed, so they are indented in the same format as shown in the statement above. Personally, I like the line continuations within parentheses to be indented to the start of the parentheses. As for line continuations not within parentheses, I'm indifferent to indenting by one or two tabs.

What would you recommend for this PR? It seems like @geogunow's proposal is to use 2 tabs regardless of whether the continuation is within parentheses - is that correct?